napari/layers/image/_async_image.py

```diff
@@ -586,101 +559,107 @@ def _set_view_slice(self):
             )
             indices[not_disp] = downsampled_indices
 
-            self._load_multi_scale(indices, level)
+            thumbnail_source = self.data[self._thumbnail_level][tuple(indices)]
+
+            self._load_async(image_indices, image, thumbnail_source)
         else:
-            self._create_image_slice()
-            self._load_single_scale()
+            self._transforms['tile2data'].scale = np.ones(self.dims.ndim)
+            image_indices = self.dims.indices
+            image = self.data[image_indices]
 
-    def _load_multi_scale(self, indices, level: int) -> None:
-        """Load multi-scale image and thumbnail_source.
-        """
-        if not self._slice:
-            self._create_image_slice()
-
-        if np.all(self._slice.current_indices == indices):
-            return  # already showing the right slice or its being loaded
-
-        # Ask for the image and the lower resolution thumbnail_source.
-        image_level = self.data[level]
-        thumbnail_level = self.data[self._thumbnail_level]
-        chunks = {
-            'image': image_level[tuple(indices)],
-            'thumbnail_source': thumbnail_level[tuple(indices)],
-        }
-        request = chunk_loader.create_request(self, indices, chunks)
+            # For single-scale we don't request a separate thumbnail_source
+            # from the ChunkLoader because in ImageSlice.chunk_loaded we
+            # call request.thumbnail_source() and it knows to just use the
+            # image itself is there is no explicit thumbnail_source.
+            thumbnail_source = None
 
-        # Load the chunks. This could load them synchronously right here in
-        # the GUI thread or it could queue up a request for a worker thread
-        # or process and self.chunk_loaded() will be called later.
-        satisfied_request = self._slice.load_chunk(request)
-        if satisfied_request is not None:
-            self.chunk_loaded(satisfied_request)
+        self._load_async(image_indices, image, thumbnail_source)
+
+    def _load_async(self, image_indices, image, thumbnail_source=None):
+        """Request the ChunkLoader to load this image and maybe thumbnail source.
 
-    def _load_single_scale(self) -> None:
-        """Load non-multiscale image.
+        The load will happen synchronously if:
+        1) The array is already an ndarray.
+        2) We are in synchronous mode.
+        3) The array was found in the ChunkCache.
```

Comment:
It should be made clear whether this is an "and" or an "or". From context I guess "or". =)

```suggestion
        1) The array is already an ndarray;
        2) We are in synchronous mode; or,
        3) The array was found in the ChunkCache.
```

(alternatively the top line could say "if any of the following conditions holds:") (or you could make both changes)

Reply:
Good point. I totally rewrote it, see [new version here](https://github.com/pwinston/napari/blob/feature/chunk-loader-2/napari/utils/chunk/_loader.py#L178-L187)